LPS-205940 Add functional tests for editor config contributor CET FF functionality

markocikos opened the pull request:
### References

- [LPS-187309 Apply editor config contributor client extensions to CKEditor](https://issues.liferay.com/browse/LPS-187309)

### What is the goal of this PR?

My initial go at Playwright. This PR is just a simple test on how FF affects option in CET menu. In a followups, we should be able to reasonably easily test creation of new CET, edit and delete. Application using a workflow sample will be a challenge.

Technical notes in comments.

### What does it look like?

![image](https://github.com/liferay-frontend/liferay-portal/assets/5435511/85f3b51d-226a-482d-a8cb-c0526dad67a6)

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-39.liferay.com/job/test-portal-source-format/5158/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-39.liferay.com/userContent/jobs/test-portal-source-format/builds/5158/jenkins-report.html">here</a>.</b></p>

markocikos commented on modules/test/playwright/fixtures/clientExtensionsPageTest.ts:
```diff
@@ -0,0 +1,22 @@
+	},
+});
+
+export {clientExtensionsPageTest};
```
I applied naming conventions to files that we [discussed in Slack](https://liferay.slack.com/archives/C06A59K5D46/p1704885957892329). I didn't apply it to existing tests, I can do it in a dedicated task.

izaera replied:
Oh, I forgot about that discusion. Then all my comments in https://github.com/liferay-frontend/liferay-portal/pull/3924 regarding naming are wrong :face_with_head_bandage:  (CC: @dsanz @chestofwonders)

However, unless we first rename everything to follow the conventions and send it to Brian, he will most probably refuse having two naming conventions merged to master... :shrug:

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 4 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/0629149307c6dae0e5a19c14b56365f90013ffeb">0629149307c6dae0e5a19c14b56365f90013ffeb</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/markocikos/liferay-portal/tree/LPS-205940">LPS-205940</a><br/>Branch GIT ID: <a href="https://github.com/markocikos/liferay-portal/commit/3761cf464e082752b9b625a324d3065822f913bc">3761cf464e082752b9b625a324d3065822f913bc</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-39.liferay.com/job/test-portal-source-format/5158/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-39.liferay.com/userContent/jobs/test-portal-source-format/builds/5158/jenkins-report.html">here</a>.</h5></details></html>

markocikos commented on modules/test/playwright/fixtures/clientExtensionsPageTest.ts:
```diff
@@ -0,0 +1,22 @@
+	clientExtensionsPage: async ({page}, use) => {
+		const clientExtensionsPage = new ClientExtensionsPage(page);
+
+		await clientExtensionsPage.goto();
```
My intent here is to set page as a fixture, so that all tests in `editorConfigContributor.spec.ts` run on the Application Menu > Client Extensions page. I am not sure if there is a cleaner way to achieve this.

izaera replied:
I'd say you can use beforeEach in the test itself, like it is used here -> https://playwright.dev/docs/test-timeouts#change-timeout-from-a-beforeeach-hook

The fixture is OK to inject the CX page, like we do with some other pages, but navigating to the page is usually done in the tests themselves, since it is a navigation on which the tests depend and the more explicit the process in the tests the better for the future.

If we hide too much in the fixtures it can be very difficult to trace what tests are really doing.

In fact, if a test depended on two fixtures providing two different pages, both would fight to go to their page, so this is clearly wrong, I think...

markocikos replied:
But, what you are suggesting negates the whole point of page fixtures. See https://playwright.dev/docs/test-fixtures#creating-a-fixture, they are using it in a similar way as in this PR:
```js
export const test = base.extend<MyFixtures>({
  todoPage: async ({ page }, use) => {
    // Set up the fixture.
    const todoPage = new TodoPage(page);
    await todoPage.goto();

    // Use the fixture value in the test.
    await use(todoPage);
  },
});
```

This is not only about convenience in tests, but also performance. Navigation is an expensive action, adding couple of seconds to each test. It will quickly pile up.

Difficulty to trace is a valid concern, but we can somewhat offset it with naming conventions for page names (`clientExtensionsPage` in this case).

izaera replied:
Mmm, you are right in the sense that you can only have one starting page in a test, which would be provided by the fixture and would set your test in a defined starting place.

However, what if you then navigate to another page in that same test?  How's the canonical way of using that new page's POM (page object model)? Do you instantiate it directly in the test? Encapsulate it inside the home page of the test as Miguel was doing in his PR? ... :thinking:

izaera replied:
OK, reading your link now I see that they do the navigation in the test file itself, even though they do it inside a composable fixture that provides a todoPage and a settingsPage.

So my point is still valid, I think. I'm requesting a separation of concerns in this case bc I assume that everything outside of tests (in this case your fixture inside the `fixtures` directory) can be reused by any test anywhere, thus it should be effect free.

So it's a matter of how we organize code in our scenario, not a problem of navigating inside a fixture which, as you said, it's a valid action if done in the proper place.

markocikos replied:
Fixtures are run with each test, repeating the navigation. To confirm this, I added to the end of first test:
```js
await clientExtensionsPage.page.goto('/o/api');
```
![image](https://github.com/liferay-frontend/liferay-portal/assets/5435511/0163ce98-b2fe-401c-9258-f8d70cfd6629)

The second one completed successfully, starting test from home page:
![image](https://github.com/liferay-frontend/liferay-portal/assets/5435511/5aef99eb-76b4-4c04-9a8e-72713285aa38)


What's the difference between this and putting navigation in each test, or `beforeEach`?

markocikos replied:
> OK, reading your link now I see that they do the navigation in the test file itself, even though they do it inside a composable fixture that provides a todoPage and a settingsPage.

What do you mean? :thinking: I don't see a difference between the sample and PR, both navigation are in fixture definition, in `test.extend`, in separate files from tests. The only difference is they also add reference to another page, `settingsPage`.

izaera replied:
> What's the difference between this and putting navigation in each test, or beforeEach?

A conceptual difference, even though both setups work.

If you put the navigation in the fixture and then inject the fixture in the test and don't navigate in the test you are assuming that the fixture contract is "when I enter the test execution the test browser will be in a certain page because of the fixture".

But now you can create another fixture that injects another page and navigates to another page, and then make your test depend on it too and in that case your test + the contracts from the two fixtures will be inconsistent, because: the test won't navigate anywhere, fixture A will "guarantee" it puts you in page X, fixture B  will "guarantee" that it puts you in Y page.

markocikos replied:
> But now you can create another fixture that injects another page and navigates to another page, and then make your test depend on it too and in that case your test + the contracts from the two fixtures will be inconsistent, because: the test won't navigate anywhere, fixture A will "guarantee" it puts you in page X, fixture B will "guarantee" that it puts you in Y page.

Yeah, that's a good point! I guess my mental model was of a unique page fixture, that makes no sense to mix with other fixtures. But if we want to make it reusable by independent sets of tests, as a way to expose page-specific locators etc, then internal navigation would clash.

I'll make the change.

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-39.liferay.com/job/test-portal-source-format/5158/">test-portal-source-format#5158</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-39.liferay.com/userContent/jobs/test-portal-source-format/builds/5158/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/3927">liferay-frontend#3927</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2486836179">[master] ci:test:sf - markocikos &gt; liferay-frontend - PR#3927 - 2024-01-11[03:22:12]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>2486836179<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-39.liferay.com/job/publish-testray-report/18924/">publish-testray-report#18924</a><br/></div></div>

markocikos commented on modules/test/playwright/tests/client-extension-web/editorConfigContributor.spec.ts:
```diff
@@ -0,0 +1,52 @@
+export const test = mergeTests(apiHelpersTest, clientExtensionsPageTest);
+
+test('Editor Config Contributor CET is disabled if FF is set to false', async ({
+	_apiHelpers,
+	clientExtensionsPage,
```
It was odd to me to see `_` prefix in a prop. I'm assuming no underscore is also another convention that we can apply for consistency.

izaera replied:
Yeah, I don't like it either. And the playwright docs don't use it. Don't know why it is used... :thinking:

markocikos commented on modules/test/playwright/tests/client-extension-web/editorConfigContributor.spec.ts:
```diff
@@ -0,0 +1,52 @@
+	_apiHelpers: any;
+	clientExtensionsPage: ClientExtensionsPage;
+}) => {
+	await _apiHelpers.featureFlag.updateFeatureFlag('LPS-186870', true);
+
+	await clientExtensionsPage.page.reload();
```
I was considering adding this to a `beforeEach`. We might want to do that if there will be more tests in this file. This is something that we need to be careful about, since `updateFeatureFlag` changes state for all following tests.

We may also want to think of some global way to set FFs, as managing FFs with fetch requests will poorly affect performance, no matter how we do it.

izaera replied:
The FF problems would be avoided with the FF fixture.

However it's not clear how we could boost performance because we would need set the FF on and off per each test, unless the executor of the playwright tests (CI, user, etc.) guarantees that only tests using the same FFs combinations are accessing the test server. This could be done theoretically by inspecting the FF fixtures the test declare, but we are far from implementing it yet.

In any case, since it would be possible as a future optimization, I would not worry about performance now but prefer to do it "right" and evolvable.

markocikos commented on modules/test/playwright/helpers/FeatureFlagApiHelper.ts:
```diff
@@ -13,7 +13,6 @@ export class FeatureFlagApiHelper {
 	}
 
 	async updateFeatureFlag(key: string, enabled: boolean) {
-		await this.page.goto('/');
```
I removed navigation here as it negates usage of page fixtures. It was unexpected to me, and needlessly worsened performance.

The downside it that there needs to be at least one navigation before `updateFeatureFlag` can be called. If not, we get a cryptic error:
![image](https://github.com/liferay-frontend/liferay-portal/assets/5435511/57e5a438-abc1-434c-9c1a-1c664882f340)

This is why I added the order change in `objectsFolder.spec.ts`

izaera replied:
IMO FF should be fixtures so that each test declares which FFs it needs and they are automagically set on/off by our testing architecture.

izaera replied:
Oh, I saw just now that the FF is turned on/off with a fetch, which is kind-of headless....

Maybe we can simply navigate to the home page when the test starts (in a global setup) and, that way, guarantee that fetch is going to work no matter where the test is.

markocikos replied:
> Maybe we can simply navigate to the home page when the test starts (in a global setup) and, that way, guarantee that fetch is going to work no matter where the test is.

Unfortunately, it doesn't seem to work that way. I expected home page would be "active" after login, but it's not. I am not sure why. Login state is preserved, but looks like not everything else related to navigation :thinking:

izaera replied:
Yeah, the current login is just loading the cookies into the context of each test but that doesn't mean the browser will have the `Liferay` object defined, obviously, because no HTML is being interpreted.

In other words, the current login is like opening a browser with prepopulated cookies (like when you click "remember me"), not like going to the home page.

markocikos replied:
So, we could add a fixure that navigates to home page, and have it composed with all tests, but:
1. That will result in quite a lot of boilerplate.
2. It's still not clear if want navigation in fixtures, see https://github.com/liferay-frontend/liferay-portal/pull/3927#discussion_r1448705875. If we implement it like this, we would need to make sure nobody adds navigation in fixtures. If a dev does it, it may result in flaky behavior, as the order of navigation would be uncertain.

I don't see how to implement this in a way to keep logic under the hood. One way or another, devs will need to know how to work around this odd quirk.

izaera replied:
I've fixed it in my login_fixture PR, in fact, see this block of code -> https://github.com/liferay/liferay-portal/commit/68586db716a18e4ef2f05486c05bff8043a026c2#diff-0dc401334992febbdd325750039c0a124f97e88116f031f78b6c7eef0384aba6R58-R67

What I do is trying to invoke `fetch` and if it fails with `ReferenceError`, go to page `/` and retry.

Since the code is encapsulated in a fixture it's a "clean" solution, meaning that the tests don't need to bother with it.

I tried to do it in the setup but it's not possible, because setup only runs once and the navigation to / needs to be done for each `Page`.

izaera replied:
I know I'm navigating inside a fixture but in this case I think it shouldn't be a problem because I only navigate in case the browser is still empty (nothing has been loaded yet) and the only side effect will be that the test asking for the fixture finds the home page preloaded when it enters which should be inocuous.

izaera replied:
In other words, navigating inside a fixture is not a problem. The problem is assuming (from a test) that because you asked for a fixture, the BrowerContext will be in a defined location.

Tests should navigate to the proper initial place as their very first action.

markocikos replied:
> I've fixed it in my login_fixture PR, in fact, see this block of code

Looks good, but I still have strong doubts abut performance. It's nothing specific to the fixture, but comes from the fact that, one way or another, we need to make a ton of requests, one per FF. I'd much prefer to see an endpoint that can handle a list of FFs and set them all at once. But, that's out of scope here. In the case of this PR, I think we can push as it as, with FFs within tests. Once the fixture approach is ready, we can refactor.

izaera replied:
>  I'd much prefer to see an endpoint that can handle a list of FFs and set them all at once.

That's definitely possible with the proposed code structure so I'm OK. I mean, I don't want to make it fast at this point, just make it evolvable, meaning we don't need to touch any test in the future if we want to optimize the code and, for example, turn all FFs in one single request, as you say.

markocikos replied:
After thinking on this, I figured the performance issue is not as bad as I initially though. It was a reflex reaction to impact of requests on standard UI, with interaction between server and browser anywhere in the world. For tests, it's always a local request, so it should be reasonably fast. It is still a performance issue, but not a severe one, something we can push down in priority.

markocikos replied:
I removed code changing how FFs are set from PR. We're using `updateFeatureFlag` util as it is, until fixure FF is available.

markocikos commented:
@izaera I force pushed:
- I moved navigation from fixture to tests.
- I left in removal of navigation from FF setter, as well as impacts in tests. I believe the logic should be the same for utilities as well as fixtures: we don't want additional sideeffects in utilities, such as navigation. Once your fixture-based solution for FFs kicks in, we can refactor.
- I extended naming conventions to all existing tests, including file names and props prefixes
- I extended TS in all existing tests

markocikos commented:
@izaera Force-pushed, with several significant changes:
- Tests now fully cover creation, editing and deletion of editor CX.
- I removed changes in FF logic, the removal of internal navigation. We're using util as it is, until FF fixture is ready.
- I moved fixtures and pages internally, as discussed in https://liferay.slack.com/archives/C06A59K5D46/p1705411600313329

Some technical notes inline.

markocikos commented on modules/test/playwright/tests/client-extension-web/editorConfigContributor.spec.ts:
```diff
@@ -0,0 +1,97 @@
+
+	await clientExtensionsPage.itemEditButton.click();
+
+	const sampleName2 = 'Sample Name 2';
+
+	await newEditorConfigContributorPage.nameInput.fill(sampleName2);
+
+	await newEditorConfigContributorPage.publishButton.click();
```
This part is flaky. Relevant Slack discussion: https://liferay.slack.com/archives/C06A59K5D46/p1705504871505079

markocikos commented on modules/test/playwright/tests/client-extension-web/pages/ClientExtensionsPage.ts:
```diff
@@ -0,0 +1,46 @@
+			name: 'Add Editor Config Contributor',
+		});
+
+		this.itemActionsToggleButton = page.getByRole('button', {
+			name: 'Actions',
+		});
+
+		this.itemDeleteButton = page.getByRole('menuitem', {
+			name: 'Delete',
+		});
+		this.itemEditButton = page.getByRole('menuitem', {
+			name: 'Edit',
+		});
```
This is assuming that no other client extensions were created previous to this test. If there are other CXs, this will fail. The test creates a CX entry and deletes it, so it it repeatable, unless it fails.

There are two ways this can be improved, but I'm not sure they are worth doing:
1. We could find the exact CX created by the test. This complicates locator, and also needs to work with pagination.
2. We could unconditionally clean up all CXs, even if tests fail.

izaera replied:
Mmm, why don't you turn the fields into methods and parameterize them? :thinking: 

Eg: change `itemEditButton` to `getItemEditButton(cxName: string): Locator` so that the caller (which is who knows the CX to be used) can specify which row to click. Would it work?

There may be a problem with pagination but that would be "your problem" in the sense that it is guaranteed that when tests start the DB is clean so you can assume there will be no CX unless other tests in your project create them and don't delete them upon exit (that's why I said it's "your" problem, because it is circumscribed to your tests only)

markocikos replied:
Yep, something like a parameterized may make sense, but it's not really necessary for this test. Later on, if we need to create multiple CXs for a test, we can do it. But for now, it's an unnecessary complication.

markocikos commented:
@izaera @dsanz Can we push this? The test is probably not ideal, but I think it is good enough for now, we can improve on it later on.

izaera commented:
LGTM, however I left a suggestion for improvement that doesn't seem to complex to do (but I leave addressing it in this PR up to you)

I agree we can do better, but then we will never forward anything as everything can be done better :sweat_smile:

markocikos commented:
ci:test:sf

markocikos commented:
ci:test:playwright

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-9.liferay.com/job/test-portal-source-format/4995/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-9.liferay.com/userContent/jobs/test-portal-source-format/builds/4995/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 6 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/2c154298829d78ac88b7410631f2f8c2c2f653ac">2c154298829d78ac88b7410631f2f8c2c2f653ac</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/markocikos/liferay-portal/tree/LPS-205940">LPS-205940</a><br/>Branch GIT ID: <a href="https://github.com/markocikos/liferay-portal/commit/215c15b9044c371b125a13b1e94bf4f474d02181">215c15b9044c371b125a13b1e94bf4f474d02181</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-9.liferay.com/job/test-portal-source-format/4995/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-9.liferay.com/userContent/jobs/test-portal-source-format/builds/4995/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-9.liferay.com/job/test-portal-source-format/4995/">test-portal-source-format#4995</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-9.liferay.com/userContent/jobs/test-portal-source-format/builds/4995/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/3927">liferay-frontend#3927</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2499150624">[master] ci:test:sf - markocikos &gt; liferay-frontend - PR#3927 - 2024-01-19[02:07:02]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>2499150624<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-9.liferay.com/job/publish-testray-report/5730/">publish-testray-report#5730</a><br/></div></div>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>playwright</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest(master)/12550/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-26.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/12550/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:playwright - 0 out of 3 jobs passed in 54 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/2c154298829d78ac88b7410631f2f8c2c2f653ac">2c154298829d78ac88b7410631f2f8c2c2f653ac</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/2c154298829d78ac88b7410631f2f8c2c2f653ac">2c154298829d78ac88b7410631f2f8c2c2f653ac</a><br/>Jenkins Build URL: <a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2497476218">EE Development Acceptance (master) - 316 - 2024-01-18[16:43:57]</a></p><details><summary><strong>ci:test:playwright - 0 out of 3 jobs PASSED</strong></summary><div><h4>3 Failed Jobs:</h4><ul><li><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest(master)/12550/"><strike>test-portal-acceptance-pullrequest(master)</strike></a></li><li>playwright-js-tomcat90-mysql57-jdk8<ul><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/325699/"><strike>playwright-js-tomcat90-mysql57-jdk8/0/0</strike></a></li><li><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/876538/"><strike>playwright-js-tomcat90-mysql57-jdk8/0/1</strike></a></li></ul></li></ul></div></details><h5>For more details click <a href="https://test-1-26.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/12550/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/325699//consoleText">playwright-js-tomcat90-mysql57-jdk8/0/0</a> - <a href="https://testray.liferay.com/reports/production/logs/2024-01/test-1-26/test-portal-acceptance-pullrequest(master)/12550/playwright-js-tomcat90-mysql57-jdk8/0/0/playwright-report/index.html">Playwright Report</a><ol><li><details><summary><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/325699/testReport/export-import-web/import.spec/ts">export-import-web/import.spec.ts</a></summary><div><ol><li><a href="https://test-1-15.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/325699//testReport/export-import-web/import.spec/ts/can_import_a_folder_with_document_type_restrictions_and_workflow">can import a folder with document type restrictions and workflow</a></li></ol></div></details></li></ol></div></li><li><div><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/876538//consoleText">playwright-js-tomcat90-mysql57-jdk8/0/1</a> - <a href="https://testray.liferay.com/reports/production/logs/2024-01/test-1-26/test-portal-acceptance-pullrequest(master)/12550/playwright-js-tomcat90-mysql57-jdk8/0/1/playwright-report/index.html">Playwright Report</a><ol><li><details><summary><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/876538/testReport/export-import-web/import.spec/ts">export-import-web/import.spec.ts</a></summary><div><ol><li><a href="https://test-1-30.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/876538//testReport/export-import-web/import.spec/ts/can_import_a_folder_with_document_type_restrictions_and_workflow">can import a folder with document type restrictions and workflow</a></li></ol></div></details></li></ol></div></li></ol><h4>For upstream results, click <a href="https://test-1-40.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">here</a>.</h4></details></html><h5>Test bundle downloads:</h5><ul><li><a href="https://files.liferay.com/private/bundles/test-1-26/jobs/test-portal-acceptance-pullrequest(master)/builds/12550/liferay-portal-bundle-tomcat.tar.gz">liferay-portal-bundle-tomcat.tar.gz</a> (mirrors: <a href="http://mirrors.dlc.liferay.com/files.liferay.com/private/bundles/test-1-26/jobs/test-portal-acceptance-pullrequest(master)/builds/12550/liferay-portal-bundle-tomcat.tar.gz">dlc</a>, <a href="http://mirrors.lax.liferay.com/files.liferay.com/private/bundles/test-1-26/jobs/test-portal-acceptance-pullrequest(master)/builds/12550/liferay-portal-bundle-tomcat.tar.gz">lax</a>)</li></ul>

markocikos commented:
Failures in https://github.com/liferay-frontend/liferay-portal/pull/3927#issuecomment-1900214047 look unrelated to me.

markocikos commented:
ci:forward

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

The pull request will automatically be forwarded to the user `brianchandotcom` If the following test suites pass:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable**

liferay-continuous-integration commented:
Skipping previously passed test suites:
- `ci:test:relevant`
- `ci:test:sf`
- `ci:test:stable`

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest(master)/12550/">test-portal-acceptance-pullrequest(master)#12550</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-26.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/12550/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>playwright<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-frontend/liferay-portal/pull/3927">liferay-frontend#3927</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2499647449">[master] ci:test:playwright - markocikos &gt; liferay-frontend - PR#3927 - 2024-01-19[02:19:20]</a><br/></div><div><strong>Testray Build ID:</strong><span></span>2499647449<br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-26.liferay.com/job/publish-testray-report/29216/">publish-testray-report#29216</a><br/></div></div>

liferay-continuous-integration commented:
All required test suite(s) passed.
Forwarding pull request to `brianchandotcom`.
[Console](https://test-1-27.liferay.com/job/forward-pullrequest/1208//consoleFull)

liferay-continuous-integration commented:
Pull request has been successfully forwarded to  https://github.com/brianchandotcom/liferay-portal/pull/146409
[Console](https://test-1-27.liferay.com/job/forward-pullrequest/1208//consoleFull)
